### PR TITLE
Updated stats module to follow latest doc style guide.

### PR DESCRIPTION
**Pull request by Soumi7**
# Updates sympy.stats to follow latest docstring style guide

#### References to other Issues or PRs

#19591

#### Brief description of what is fixed or changed
- Required sections in :
  - sympy/stats/crv.py
  - sympy/stats/crv_types.py
  - sympy/stats/drv.py
  - sympy/stats/drv_types.py
  - sympy/stats/matrix_distributions.py
  - sympy/stats/rv.py
  - sympy/stats/rv_interface.py
  - sympy/stats/stochastic_process.py
  - sympy/stats/stochastic_process_types.py
  - sympy/stats/symbolic_multivariate_probability.py
 
#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/stats/crv.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def as_boolean(self):
 class ConditionalContinuousDomain(ContinuousDomain, ConditionalDomain):
     """
     A domain with continuous support that has been further restricted by a
-    condition such as x > 3
+    condition such as x > 3.
```
Comment (sylee957) [inline sympy/stats/crv.py:81]:
```suggestion
    condition such as $x > 3$.
```
I'd try to abbreviate these formulas with LaTeX or inline literals for this or any other similar occurrences of this.
The golden rule I'd use is
1. If the symbols are short (`x, y, x_1, ...`) and have well-defined LaTeX syntax, try to use LaTeX `$ $`
2. If the symbols are long or use everyday words (`foo, bar, foo_bar, ...`), try to use inline literal ` `` `` `
3. If you are not confident with, try to use inline literal ` `` `` ` than experimenting with LaTeX

Comment (Soumi7) [reply sympy/stats/crv.py]: Yes. Will do this.

File: sympy/stats/crv_types.py [hunk 2]
```diff
@@ -864,8 +867,10 @@ def ChiNoncentral(name, k, l):
     Parameters
     ==========
 
-    k : A positive Integer, `k > 0`, the number of degrees of freedom
-    lambda : Real number, `\lambda > 0`, Shift parameter
+    k : A positive Integer, ``k > 0``
```
Comment (sylee957) [inline sympy/stats/crv_types.py:870]:
```suggestion
    k : A positive Integer, $k > 0$
```

File: sympy/stats/crv_types.py [hunk 3]
```diff
@@ -1177,13 +1195,16 @@ def ExGaussian(name, mean, std, rate):
...
+    ===========
+
     The density of the exponentially modified Gaussian distribution is given by
 
     .. math::
         f(x) := \frac{\lambda}{2}e^{\frac{\lambda}{2}(2\mu+\lambda\sigma^2-2x)}
             \text{erfc}(\frac{\mu + \lambda\sigma^2 - x}{\sqrt{2}\sigma})
 
-    with `x > 0`. Note that the expected value is `1/\lambda`.
+    with ``x > 0``. Note that the expected value is `1/\lambda`.
```
Comment (sylee957) [inline sympy/stats/crv_types.py:1207]:
```suggestion
    with $x > 0$. Note that the expected value is `1/\lambda`.
```

File: sympy/stats/crv_types.py [hunk 4]
```diff
@@ -1278,12 +1299,15 @@ def Exponential(name, rate):
...
+    Explanation
+    ===========
+
     The density of the exponential distribution is given by
 
     .. math::
         f(x) := \lambda \exp(-\lambda x)
 
-    with `x > 0`. Note that the expected value is `1/\lambda`.
+    with ``x > 0``. Note that the expected value is `1/\lambda`.
```
Comment (sylee957) [inline sympy/stats/crv_types.py:1310]:
```suggestion
    with $x > 0$. Note that the expected value is `1/\lambda`.
```

File: sympy/stats/crv_types.py [hunk 5]
```diff
@@ -4033,8 +4154,11 @@ def UniformSum(name, n):
     r"""
     Create a continuous random variable with an Irwin-Hall distribution.
 
+    Explanation
+    ===========
+
     The probability distribution function depends on a single parameter
-    `n` which is an integer.
+    ``n`` which is an integer.
```
Comment (sylee957) [inline sympy/stats/crv_types.py:4161]:
```suggestion
    $n$ which is an integer.
```

File: sympy/stats/crv_types.py [hunk 6]
```diff
@@ -3012,8 +3103,8 @@ def GaussianInverse(name, mean, shape):
     Parameters
     ==========
 
-    mu : Positive number representing the mean
-    lambda : Positive number representing the shape parameter
+    mu : Positive number representing the mean.
```
Comment (sylee957) [inline sympy/stats/crv_types.py:3106]:
```suggestion
    mu
        Positive number representing the mean.
```

File: sympy/stats/crv_types.py [hunk 7]
```diff
@@ -3012,8 +3103,8 @@ def GaussianInverse(name, mean, shape):
...
+    lambda : Positive number representing the shape parameter.
```
Comment (sylee957) [inline sympy/stats/crv_types.py:3107]:
```suggestion
    lambda : 
        Positive number representing the shape parameter.
```

File: sympy/stats/crv_types.py [hunk 8]
```diff
@@ -2484,8 +2554,10 @@ def LogNormal(name, mean, std):
     Parameters
     ==========
 
-    mu : Real number, the log-scale
-    sigma : Real number, :math:`\sigma^2 > 0` a shape
+    mu : Real number
+        The log-scale.
+    sigma : Real number, :math:`\sigma^2 > 0`
+        A shape.
```
Comment (sylee957) [inline sympy/stats/crv_types.py:2560]:
```suggestion
    sigma : Real number
        A shape. ($\sigma^2 > 0$)
```

File: sympy/stats/crv_types.py (hunk 2 above)
Comment (Soumi7) [reply sympy/stats/crv_types.py]: Changed.

File: sympy/stats/crv_types.py (hunk 3 above)
Comment (Soumi7) [reply sympy/stats/crv_types.py]: Changed.

File: sympy/stats/crv_types.py (hunk 4 above)
Comment (Soumi7) [reply sympy/stats/crv_types.py]: Changed.

File: sympy/stats/crv_types.py (hunk 5 above)
Comment (Soumi7) [reply sympy/stats/crv_types.py:4161]: Changed.

File: sympy/stats/crv_types.py (hunk 7 above)
Comment (Soumi7) [reply sympy/stats/crv_types.py:3107]: Changed.

File: sympy/stats/crv_types.py (hunk 6 above)
Comment (Soumi7) [reply sympy/stats/crv_types.py:3106]: Changed.

File: sympy/stats/crv_types.py (hunk 8 above)
Comment (Soumi7) [reply sympy/stats/crv_types.py:2560]: Changed.

**Pull request merged**